packages/communication/src/index.ts

```diff
@@ -0,0 +1,22 @@
+import Web3 from 'web3'
+import { NodeCommunicationWrapper } from './node-communication-wrapper'
+import { ReadOnlyWallet } from './wallet'
+export * from './abi'
+export * from './commons'
+export * from './node-communication-wrapper'
+export * from './utils/abi-utils'
+export * from './utils/celo-transaction-object'
+export * from './utils/rpc-caller'
+export * from './utils/tx-result'
+export * from './wallet'
+
+/**
+ * Creates a new instance of `CommunicationWrapper` with a web3 instance
+ * @param web3 Web3 instance
+ */
+export function newCommunicationWrapperFromWeb3(web3: Web3, wallet?: ReadOnlyWallet) {
```

Comment:
yikes! Similar to what I wrote above, instead of this factory pattern how about we let users just call `new` and we do that currentProvider check in the constructor?

If we really want the factories still, maybe from `fromWeb3` would suffice.